### PR TITLE
Scope baggage/context interaction functionality

### PR DESCRIPTION
**Pull request by MrAlias**
[The Baggage API states](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.0.1/specification/baggage/api.md#overview) that it can be implemented by providing functionality interacting solely with the `Context`. If an implementation chooses this option it follows that it should not be required to provide access to a Baggage object. This updates the Context Interaction section of the Baggage API to clarify this caveat and makes the caveat meaning clearer in the Overview section.

### CODE REVIEW

File: specification/baggage/api.md [hunk 1]
```diff
@@ -101,8 +101,8 @@ REQUIRED parameters:
 This section defines all operations within the Baggage API that interact with
 the [`Context`](../context/context.md).
 
-The API MUST provide the following functionality to interact with a `Context`
-instance:
+If an implementation of this API does not interact purely via the `Context`, it
```
[the remark this answers is not in the document]
Comment (MrAlias) [reply specification/baggage/api.md:104]: I'm in favor of your suggestion, but I was copying language from the Overview section of the specification that stated the option with these words.

Do you think it is worth updating that section to match? I'm okay with this as well. My original intention was to keep the change as small as possible, but if it isn't clarifying things it isn't worth it either.

[the remark this answers is not in the document]
Comment (yurishkuro) [reply specification/baggage/api.md:104]: I just re-read the above, and suggest the following:

1) change L37-38 to 
```
Depending on language idioms, a language API MAY implement these functions by
interacting with the baggage directly via the `Context` object.
```

2) this line:

```suggestion
If an implementation of this API does not operate directly on the `Context`, it
```

File: specification/baggage/api.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ The Baggage API consists of:
 The functions described here are one way to approach interacting with the
 `Baggage` via having struct/object that represents the entire Baggage content.
 Depending on language idioms, a language API MAY implement these functions by
-providing the defined functionality interacting purely via the `Context`.
+interacting with the `Baggage` via the `Context` directly.
```
Comment (yurishkuro) [inline specification/baggage/api.md:38]:
```suggestion
interacting with the Baggage via the `Context` directly.
```
(not referring to the `Baggage` type, but to the concept of Baggage)

**Review by yurishkuro — APPROVED** [review]

**Review by SergeyKanzhelev — APPROVED** [review]

**Review by reyang — APPROVED** [review]

File: specification/baggage/api.md (hunk 2 above)
Comment (MrAlias) [reply specification/baggage/api.md:38]: ah, SGTM :+1:

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**